simple-loadflow/src/main/java/com/powsybl/loadflow/simple/ac/ClosedBranchActiveFlowEquationTerm.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.loadflow.simple.ac;
+
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.loadflow.simple.equations.EquationContext;
+import com.powsybl.loadflow.simple.equations.EquationType;
+import com.powsybl.loadflow.simple.equations.Variable;
+import com.powsybl.loadflow.simple.network.BranchCharacteristics;
+import com.powsybl.loadflow.simple.network.NetworkContext;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class ClosedBranchActiveFlowEquationTerm extends AbstractClosedBranchAcFlowEquationTerm {
+
+    public ClosedBranchActiveFlowEquationTerm(ClosedBranchAcContext branchContext, Branch.Side side, Bus bus1, Bus bus2,
+                                              NetworkContext networkContext, EquationContext equationContext) {
+        super(branchContext, side, bus1, bus2, EquationType.BUS_P, networkContext, equationContext);
+    }
+
+    @Override
+    public double eval(double[] x) {
+        Objects.requireNonNull(x);
+        double v1 = x[branchContext.getV1Var().getColumn()];
+        double v2 = x[branchContext.getV2Var().getColumn()];
+        double ph1 = x[branchContext.getPh1Var().getColumn()];
+        double ph2 = x[branchContext.getPh2Var().getColumn()];
+        switch (side) {
+            case ONE:
+                return branchContext.p1(v1, v2, ph1, ph2);
+            case TWO:
+                return branchContext.p2(v1, v2, ph1, ph2);
+            default:
+                throw new IllegalStateException("Unknown side: " + side);
+        }
+    }
+
+    private double dp1(Variable variable, double v1, double v2, double ph1, double ph2) {
```

Comment:
Maybe we could make it even more clearer by having separate functions dp1dv1 etc.
Maybe the dispatching to those functions could be done in an abstract classe?

Reply:
Done for functions. Not so easy for common dispatching method. I gave a try using method references but not all dxdy have the same signature.